test/controllers/DeclarationDetailsControllerSpec.scala

```diff
@@ -0,0 +1,143 @@
+    super.beforeEach()
+
+    authorizedUser()
+    when(declarationInformationPage.apply(any(), any())(any(), any())).thenReturn(HtmlFormat.empty)
+    when(declarationDetailsPage.apply(any(), any())(any(), any())).thenReturn(HtmlFormat.empty)
+  }
+
+  override protected def afterEach(): Unit =
+    reset(queryNotificationMessageConfig, declarationInformationPage, declarationDetailsPage, mockCustomsDeclareExportsConnector)
+
+  "Display Submissions" should {
+
+    "return 200 (OK)" when {
+
+      "display declaration with notification method is invoked" when {
+
+        "QueryNotificationMessage feature flag is disabled" in {
+
+          responsesToReturn(isQueryNotificationMessageEnabled = false)
+
+          val result = controller.displayPage(actionId)(getRequest())
+          status(result) mustBe OK
+
+          verifyNoInteractions(declarationDetailsPage)
+
+          verifyPage { (submissionCaptor: ArgumentCaptor[Submission], notificationsCaptor: ArgumentCaptor[Seq[Notification]]) =>
+            verify(declarationInformationPage).apply(submissionCaptor.capture(), notificationsCaptor.capture())(any(), any())
+          }
+        }
+
+        "QueryNotificationMessage feature flag is enabled" in {
+
+          responsesToReturn(isQueryNotificationMessageEnabled = true)
+
+          val result = controller.displayPage(actionId)(getRequest())
+          status(result) mustBe OK
+
+          verifyNoInteractions(declarationInformationPage)
+
+          verifyPage { (submissionCaptor: ArgumentCaptor[Submission], notificationsCaptor: ArgumentCaptor[Seq[Notification]]) =>
+            verify(declarationDetailsPage).apply(submissionCaptor.capture(), notificationsCaptor.capture())(any(), any())
+          }
+        }
+      }
+
+      def responsesToReturn(isQueryNotificationMessageEnabled: Boolean): OngoingStubbing[Future[Seq[Notification]]] = {
+        when(queryNotificationMessageConfig.isQueryNotificationMessageEnabled).thenReturn(isQueryNotificationMessageEnabled)
+        when(mockCustomsDeclareExportsConnector.findSubmission(any())(any(), any())).thenReturn(Future.successful(Some(submission)))
+        when(mockCustomsDeclareExportsConnector.findNotifications(any())(any(), any())).thenReturn(Future.successful(Seq(notification)))
+      }
+
+      def verifyPage(callsiteToVerify: (ArgumentCaptor[Submission], ArgumentCaptor[Seq[Notification]]) => HtmlFormat.Appendable): Assertion = {
+        val submissionCaptor: ArgumentCaptor[Submission] = ArgumentCaptor.forClass(classOf[Submission])
+        val notificationsCaptor: ArgumentCaptor[Seq[Notification]] = ArgumentCaptor.forClass(classOf[Seq[Notification]])
+        callsiteToVerify(submissionCaptor, notificationsCaptor)
+        submissionCaptor.getValue mustBe submission
+        notificationsCaptor.getValue mustBe Seq(notification)
+      }
+    }
+
+    "return 303 (SEE_OTHER)" when {
+
+      "there is no submission during display Declaration with notification method" in {
```

Comment:
It might be useful to add a test scenario when there is a `Submission`, but no `Notification` related to it.